homeassistant/components/duckdns/config_flow.py

```diff
@@ -0,0 +1,106 @@
+"""Config flow for DuckDNS integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
+
+from . import update_duckdns
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("domain"): str,
+        vol.Required("access_token"): str,
+    }
+)
+
+
+class PlaceholderHub:
+    """Placeholder class to make tests pass."""
+
+    def __init__(self) -> None:
+        """Initialize."""
+
+    async def authenticate(
+        self, hass: HomeAssistant, domain: str, access_token: str
+    ) -> bool:
+        """Test if we can authenticate with the host."""
+        session = async_get_clientsession(hass)
+        return await update_duckdns(session, domain, access_token)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect."""
+    hub = PlaceholderHub()
+
+    if not await hub.authenticate(hass, data["domain"], data["access_token"]):
+        raise InvalidAuth
+
+    return {"title": "DuckDNS"}
```

Comment:
There is no real info retrieved here, the return value is also static.

Maybe just incorporate it into the flow below and remove the need for this method completely.

Reply:
Removed validate_input and return nothing.